skia-safe/examples/vulkan-window/renderer.rs

```diff
@@ -57,16 +60,22 @@ impl VulkanRenderer {
         let (swapchain, _images) = {
             // Querying the capabilities of the surface. When we create the swapchain we can only
             // pass values that are allowed by the capabilities.
-            let surface_capabilities = device
-                .physical_device()
+            let physical_device = device.physical_device();
+            let surface_capabilities = physical_device
                 .surface_capabilities(&surface, Default::default())
                 .unwrap();
 
             // Choosing the internal format that the images will have.
             let (image_format, _) = device
                 .physical_device()
                 .surface_formats(&surface, Default::default())
-                .unwrap()[0];
+                .unwrap()
+                .into_iter()
+                .find(|a| a.0 == Format::B8G8R8A8_UNORM)
```

Comment:
💯 This definitely needs to match the format requested when creating a surface. One alternative to hard-coding would be to set up [an allowlist](https://github.com/samizdatco/skia-canvas/blob/bbf3094825f85706373b64c9a5ce573e5ccf49e9/src/gpu/vulkan/mod.rs#L9) and then [step through](https://github.com/samizdatco/skia-canvas/blob/bbf3094825f85706373b64c9a5ce573e5ccf49e9/src/gpu/vulkan/renderer.rs#L122) the GPU's supported formats until finding one that matches